src/com/collection/MyLinkedList.java

```diff
@@ -0,0 +1,109 @@
+package collection;
+
+public class MyLinkedList<T> implements List<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = first;
+        size = 0;
+    }
+
+    private static class Node<T> {
+        public Node<T> prev;
+        public Node<T> next;
+        public T data;
+
+        public Node(Node<T> prev, T data, Node<T> next) {
+            this.prev = prev;
+            this.data = data;
+            this.next = next;
+        }
+    }
+
+    @Override
+    public void add(T item) {
+        Node<T> newNode = new Node<>(last, item, null);
+        if (size == 0) {
+            first = newNode;
+        }
+        last.next = newNode;
+        newNode.prev = last;
+        last = newNode;
+        size++;
+    }
+
+    @Override
+    public void remove(int i) {
+        Node<T> x = getNode(i);
+        if (x != null) {
+            remove(x);
+        }
+    }
+
+    private void remove(Node<T> node) {
+        Node<T> prev = node.prev;
+        Node<T> next = node.next;
+        if (prev.data == null) {
+            first = node.next;
+        } else if (next == null) {
+            last = node.prev;
+        } else {
+            prev.next = next;
+            next.prev = prev;
+            node.next = null;
+            node.prev = null;
+        }
+        node.data = null;
+        size--;
+
+    }
+
+    private Node<T> getNode(int index) throws NodeIndexOutBoundaryException {
```

Comment:
If NodeIndexOutBoundaryException extends from RuntimeException you don't need to write throws NodeIndexOutBoundaryException. It is an unchecked exception.